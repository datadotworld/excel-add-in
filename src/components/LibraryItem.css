/*
 * Copyright 2017 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
.item {
  background-color: #fff;
  position: relative;
  border-bottom: 1px solid #dfe3e9;
  padding: 10px 15px;
  display: flex;
  align-items: center;
  width: 100%;
  cursor: pointer;
}

.recent {
  cursor: default;
}

.date {
  font-size: 14px;
  color: gray;
  padding: 0.5rem;
  padding-left: 16px;
}

.item .center-info {
  margin-left: 15px;
  max-width: 210px;
}

.item .title {
  font-size: 14px;
  font-weight: 700;
  white-space: nowrap;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
}

.item .info {
  font-size: 12px;
  color: #8c9caf;
  white-space: nowrap;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
}

.item .add-icon {
  height: 0.9375rem;
}
.tabular-icon {
  height: 1.25rem;
}

.link-icon {
  justify-self: flex-end;
  margin-left: auto;
}

.link-icon .stroke {
  fill: #c1c1c1;
}

.resync-icon {
  fill: #517fb8;
  margin-top: 0.125rem;
  margin-left: 0.25rem;
}

.loading-border {
  width: 10px;
  height: 10px;
}

.icon-border {
  margin-left: auto;
  cursor: pointer;
}

.upload-button {
  font-size: 13px;
  color: #517fb8;
  padding-right: 0.9375rem;
  font-weight: 700;
  text-decoration: none !important;
}

.large-header {
  margin-right: auto;
}
