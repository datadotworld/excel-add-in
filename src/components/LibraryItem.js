/*
 * Copyright 2017 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedDate } from 'react-intl';
import { Image } from 'react-bootstrap';
import analytics from '../analytics';

import './LibraryItem.css';
import Icon from './icons/Icon';
const add = require('./icons/icon-add.svg');

class LibraryItem extends Component {
  static propTypes = {
    library: PropTypes.object,
    buttonText: PropTypes.string,
    buttonHandler: PropTypes.func,
    buttonLink: PropTypes.string
  };

  buttonClick = () => {
    analytics.track(
      `exceladdin.dataset.${this.props.buttonText.toLowerCase()}.click`
    );
    if (this.props.buttonLink) {
      return window.open(this.props.buttonLink);
    }
    if (this.props.buttonHandler) {
      const { library } = this.props;
      const uri = `https://data.world/${library.owner}/${library.id}`;
      this.props.buttonHandler(uri);
    }
  };

  render() {
    const { library } = this.props;
    const { isProject } = library;

    if (library.isCreate) {
      return (
        <div className="dataset" onClick={this.buttonClick}>
          <Image className="add-icon" src={add} />
          <div className="center-info">
            <div className="title">
              {`Create a new ${isProject ? 'project' : 'dataset'}`}
            </div>
          </div>
          <div className="link-icon">
            <Icon icon="angleRight" />
          </div>
        </div>
      );
    }
    return (
      <div className="dataset" onClick={this.buttonClick}>
        <Icon icon={library.isProject ? 'projectSchema' : 'datasetSchema'} />
        <div className="center-info">
          <div className="title">{library.title}</div>
          <div className="info">
            @{library.owner} &middot; Updated{' '}
            <FormattedDate
              value={library.updated}
              year="numeric"
              month="short"
              day="2-digit"
            />
          </div>
        </div>
        <div className="link-icon">
          <Icon icon="angleRight" />
        </div>
      </div>
    );
  }
}

export default LibraryItem;
