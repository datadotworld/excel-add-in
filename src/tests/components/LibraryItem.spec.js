/*
 * Copyright 2017 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

import LibraryItem from '../../components/LibraryItem';

const dataset = {
  owner: 'test',
  id: 'test',
  title: 'This is a test',
  updated: '2017-07-20T14:24:51.762Z',
  created: '2017-07-20T14:24:51.762Z'
};

it('renders dataset', () => {
  expect(
    renderer
      .create(
        <IntlProvider locale="en">
          <LibraryItem item={dataset} />
        </IntlProvider>
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders dataset - button text', () => {
  expect(
    renderer
      .create(
        <IntlProvider locale="en">
          <LibraryItem item={dataset} buttonText="test" />
        </IntlProvider>
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders dataset - button text, link', () => {
  expect(
    renderer
      .create(
        <IntlProvider locale="en">
          <LibraryItem
            item={dataset}
            buttonText="test"
            buttonLink="https://data.world"
          />
        </IntlProvider>
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders dataset - as project', () => {
  const project = Object.assign(dataset);
  project.isProject = true;

  expect(
    renderer
      .create(
        <IntlProvider locale="en">
          <LibraryItem item={project} />
        </IntlProvider>
      )
      .toJSON()
  ).toMatchSnapshot();
});
