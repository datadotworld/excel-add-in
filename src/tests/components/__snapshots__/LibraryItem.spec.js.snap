// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dataset - as project 1`] = `
<div
  className="dataset"
  onClick={[Function]}
>
  <span
    className="svg-icon "
    onClick={undefined}
  >
    <svg
      className="project-schema-icon"
      version="1.1"
      viewBox="0 0 16 16"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <path
        className="fill"
        d="M15,0H1C0.4,0,0,0.4,0,1v14c0,0.6,0.4,1,1,1h14c0.6,0,1-0.4,1-1V1C16,0.4,15.6,0,15,0z M13.6,6.3l-2.3,3.5c-0.3,0.6-1.3,0.7-1.7,0.2L7,7.5l-2.7,3.6c-0.3,0.5-1.2,0.6-1.6,0.2C2.2,11,2.1,10.1,2.4,9.7L5.9,5c0.5-0.6,1.3-0.6,1.7-0.1l2.4,2.4L11.6,5c0.3-0.6,1-0.7,1.6-0.3C13.8,5,13.9,5.7,13.6,6.3z"
      />
    </svg>
  </span>
  <div
    className="center-info"
  >
    <div
      className="title"
    >
      This is a test
    </div>
    <div
      className="info"
    >
      @
      test
       路 Updated
       
      <span>
        Jul 20, 2017
      </span>
    </div>
  </div>
  <div
    className="link-icon"
  >
    <span
      className="svg-icon "
      onClick={undefined}
    >
      <svg
        className="angle-right"
        height="40"
        viewBox="0 0 40 40"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="stroke"
          d="m26.3 21.4q0 0.3-0.2 0.5l-10.4 10.4q-0.3 0.3-0.6 0.3t-0.5-0.3l-1.1-1.1q-0.2-0.2-0.2-0.5t0.2-0.5l8.8-8.8-8.8-8.7q-0.2-0.3-0.2-0.6t0.2-0.5l1.1-1.1q0.3-0.2 0.5-0.2t0.6 0.2l10.4 10.4q0.2 0.2 0.2 0.5z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`renders dataset - button text 1`] = `
<div
  className="dataset"
  onClick={[Function]}
>
  <span
    className="svg-icon "
    onClick={undefined}
  >
    <svg
      className="dataset-schema-icon"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="fill"
        d="M15,0H1C0.4,0,0,0.4,0,1v14c0,0.6,0.4,1,1,1h14c0.6,0,1-0.4,1-1V1C16,0.4,15.6,0,15,0z M8,13.5L2.2,9.9l1.5-1L8,11.5l4.3-2.7l1.5,0.9L8,13.5z M8,9.9L2.2,6.1L8,2.5l5.8,3.6L8,9.9z"
      />
    </svg>
  </span>
  <div
    className="center-info"
  >
    <div
      className="title"
    >
      This is a test
    </div>
    <div
      className="info"
    >
      @
      test
       路 Updated
       
      <span>
        Jul 20, 2017
      </span>
    </div>
  </div>
  <div
    className="link-icon"
  >
    <span
      className="svg-icon "
      onClick={undefined}
    >
      <svg
        className="angle-right"
        height="40"
        viewBox="0 0 40 40"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="stroke"
          d="m26.3 21.4q0 0.3-0.2 0.5l-10.4 10.4q-0.3 0.3-0.6 0.3t-0.5-0.3l-1.1-1.1q-0.2-0.2-0.2-0.5t0.2-0.5l8.8-8.8-8.8-8.7q-0.2-0.3-0.2-0.6t0.2-0.5l1.1-1.1q0.3-0.2 0.5-0.2t0.6 0.2l10.4 10.4q0.2 0.2 0.2 0.5z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`renders dataset - button text, link 1`] = `
<div
  className="dataset"
  onClick={[Function]}
>
  <span
    className="svg-icon "
    onClick={undefined}
  >
    <svg
      className="dataset-schema-icon"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="fill"
        d="M15,0H1C0.4,0,0,0.4,0,1v14c0,0.6,0.4,1,1,1h14c0.6,0,1-0.4,1-1V1C16,0.4,15.6,0,15,0z M8,13.5L2.2,9.9l1.5-1L8,11.5l4.3-2.7l1.5,0.9L8,13.5z M8,9.9L2.2,6.1L8,2.5l5.8,3.6L8,9.9z"
      />
    </svg>
  </span>
  <div
    className="center-info"
  >
    <div
      className="title"
    >
      This is a test
    </div>
    <div
      className="info"
    >
      @
      test
       路 Updated
       
      <span>
        Jul 20, 2017
      </span>
    </div>
  </div>
  <div
    className="link-icon"
  >
    <span
      className="svg-icon "
      onClick={undefined}
    >
      <svg
        className="angle-right"
        height="40"
        viewBox="0 0 40 40"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="stroke"
          d="m26.3 21.4q0 0.3-0.2 0.5l-10.4 10.4q-0.3 0.3-0.6 0.3t-0.5-0.3l-1.1-1.1q-0.2-0.2-0.2-0.5t0.2-0.5l8.8-8.8-8.8-8.7q-0.2-0.3-0.2-0.6t0.2-0.5l1.1-1.1q0.3-0.2 0.5-0.2t0.6 0.2l10.4 10.4q0.2 0.2 0.2 0.5z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`renders dataset 1`] = `
<div
  className="dataset"
  onClick={[Function]}
>
  <span
    className="svg-icon "
    onClick={undefined}
  >
    <svg
      className="dataset-schema-icon"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="fill"
        d="M15,0H1C0.4,0,0,0.4,0,1v14c0,0.6,0.4,1,1,1h14c0.6,0,1-0.4,1-1V1C16,0.4,15.6,0,15,0z M8,13.5L2.2,9.9l1.5-1L8,11.5l4.3-2.7l1.5,0.9L8,13.5z M8,9.9L2.2,6.1L8,2.5l5.8,3.6L8,9.9z"
      />
    </svg>
  </span>
  <div
    className="center-info"
  >
    <div
      className="title"
    >
      This is a test
    </div>
    <div
      className="info"
    >
      @
      test
       路 Updated
       
      <span>
        Jul 20, 2017
      </span>
    </div>
  </div>
  <div
    className="link-icon"
  >
    <span
      className="svg-icon "
      onClick={undefined}
    >
      <svg
        className="angle-right"
        height="40"
        viewBox="0 0 40 40"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="stroke"
          d="m26.3 21.4q0 0.3-0.2 0.5l-10.4 10.4q-0.3 0.3-0.6 0.3t-0.5-0.3l-1.1-1.1q-0.2-0.2-0.2-0.5t0.2-0.5l8.8-8.8-8.8-8.7q-0.2-0.3-0.2-0.6t0.2-0.5l1.1-1.1q0.3-0.2 0.5-0.2t0.6 0.2l10.4 10.4q0.2 0.2 0.2 0.5z"
        />
      </svg>
    </span>
  </div>
</div>
`;
